nisapi/clean/helpers.py

```diff
@@ -307,3 +310,116 @@ def is_valid_age_groups(x: pl.Series) -> bool:
         x (pl.Series): series of age groups
     """
     return x.str.contains(r"^(\d+-\d+|\d+\+) years$").all()
+
+
+def week_ending_to_times(df: pl.LazyFrame) -> pl.LazyFrame:
+    """Convert `week_ending` to time type, time start, and time end
+
+    Args:
+        df (pl.LazyFrame): input frame with column "week_ending"
+
+    Returns:
+        pl.LazyFrame: frame without "week_ending" but with "time_type",
+          "time_start", and "time_end"
+    """
+    name = str(uuid.uuid1())
+    return (
+        df.with_columns(_week_ending_to_times_expr(pl.col("week_ending")).alias(name))
+        .unnest(name)
+        .drop("week_ending")
+        .with_columns(time_type=pl.lit("week"))
+    )
+
+
+def _week_ending_to_times_expr(week_ending: pl.Expr) -> pl.Expr:
+    week_start = week_ending.dt.offset_by("-6d")
```

Comment:
To clarify, when NIS data has a "week ending" column, we assume the week of data collection began 6 days prior, rather than the day after the previous "week ending" entry. For example, suppose two successive "week ending" entries are "11-17-2024" and "11-25-2024" (that is, one of those weird 8-day "weeks" that sometimes appear). I might have assumed the second uptake estimate is for the time period 11-18-2024 to 11-25-2024, but this code will assume the second uptake estimate is for the time period 11-19-2024 to 11-25-2024.

This will cause the daily average uptake for the second uptake estimate to be averaged over 7 days instead of 8. So it will affect the model fitting procedure slightly... but probably not enough to really matter.

Just checking that this is the behavior we want.

Reply:
Moved to #44